src/NHibernate/Async/Event/Default/DefaultLockEventListener.cs

```diff
@@ -0,0 +1,93 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using NHibernate.Engine;
+using NHibernate.Persister.Entity;
+using NHibernate.Proxy;
+
+namespace NHibernate.Event.Default
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	/// <content>
+	/// Contains generated async methods
+	/// </content>
+	public partial class DefaultLockEventListener : AbstractLockUpgradeEventListener, ILockEventListener
+	{
+		/// <summary>Handle the given lock event. </summary>
+		/// <param name="event">The lock event to be handled.</param>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		public virtual Task OnLockAsync(LockEvent @event, CancellationToken cancellationToken)
+		{
+			if (@event.Entity == null)
+			{
+				throw new NullReferenceException("attempted to lock null");
+			}
+
+			if (@event.LockMode == LockMode.Write)
+			{
+				throw new HibernateException("Invalid lock mode for lock()");
+			}
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<object>(cancellationToken);
+			}
+			async Task InternalOnLockAsync()
```

Comment:
Yes I know it which case it should use a tail function, and I am for it. I was only referring to where we put them inside their holding function. Maybe I should have written "formatting" instead of "usage".